### PR TITLE
Adding the api documents to publisher UI

### PR DESCRIPTION
**Pull request by irham0019**

### CODE REVIEW

File: components/web/apps/org.wso2.carbon.apimgt.admin.app/src/main/fragments/action-menu-policy-add/action-menu-policy-add.hbs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<div id="navbar" class="collapse navbar-collapse">
+    <ul class="nav navbar-nav">
+        <li>
+            <a title="Add" href="{{env.contextPath}}/throttling/policy-add?{{@params.policy_type}}=true">
+	                                <span class="icon fw-stack">
+	                                    <i class="fw fw-add fw-stack-1x" title={{i18n "Add Policy"}}></i>
+	                                    <i class="fw fw-circle-outline fw-stack-2x" title={{i18n "Add Policy"}}></i>
+	                                </span>
+                Add
```
Comment (harsha89) [inline components/web/apps/org.wso2.carbon.apimgt.admin.app/src/main/fragments/action-menu-policy-add/action-menu-policy-add.hbs]: Do we need to localize this?

File: components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                            <textarea  id="summary" class="required form-control" rows="3" style="width:100%" title="summary"></textarea>
+                        </div>
+                    </div>
+
+
+                </div>
+                <div class="col-sm-4">
+                    <div class="well">
+                        <div class="control-group">
+                            <label class="control-label">Type</label>
```
Comment (harsha89) [inline components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs]: Localization need to be add?

File: components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                            <ul class="nav">
+                                <li><label> <input type="radio"
+                                                   name="optionsRadios" id="optionsRadios1" value="HOWTO"
+                                                   checked="" onChange="disableInline()"
+                                                   onclick="javascript:$('#otherTypeDiv').hide('slow')" /> How To
```
Comment (harsha89) [inline components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs]: Localization need to be add?

File: components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                                </label></li>
+                                <li><label> <input type="radio"
+                                                   name="optionsRadios" id="optionsRadios2" value="SAMPLES"
+                                                   onChange="disableInline()"
```
Comment (harsha89) [inline components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs]: Localization need to be add?

File: components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                                                   onclick="javascript:$('#otherTypeDiv').hide('slow')" /> Samples & SDK
+                                </label></li>
+                                <li><label> <input type="radio"
+                                                   name="optionsRadios" id="optionsRadios3" value="PUBLIC_FORUM"
+                                                   onChange="disableInline('forum')"
+                                                   onclick="javascript:$('#otherTypeDiv').hide('slow')" /> Public Forum
```
Comment (harsha89) [inline components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs]: Localization need to be add?

File: components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+                                </label></li>
+                                <label> <input type="radio" name="optionsRadios"
+                                               id="optionsRadios4" value="SUPPORT_FORUM"
+                                               onChange="disableInline('forum')"
+                                               onclick="javascript:$('#otherTypeDiv').hide('slow')" /> Support Forum
```
Comment (harsha89) [inline components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs]: Same here

File: components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+                                </label>
+                                <li><label> <input type="radio"
+                                                   name="optionsRadios" id="optionsRadios5" value="OTHER"
+                                                   onChange="disableInline()"
+                                                   onclick="javascript:$('#otherTypeDiv').toggle('slow')" />Other (specify)
```
Comment (harsha89) [inline components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs]: Localization need to be add?

File: components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+                    <div class="well">
+                        <div class="form-group">
+                            <label class="control-label">Source</label>
+                            <ul class="nav">
+                                <li>
+                                    <label>
+                                        <input type="radio" name="optionsRadios1" id="optionsRadios6" value="INLINE" onclick="javascript:$('#sourceUrlDoc').hide('slow');$('#sourceFile').hide('slow')" checked="">Inline
+                                        <a class="icon-question-sign help_popup" help_data="inlineDoc_help" title="inlineDoc_help""></a>
+                                        <p id="inlineDoc_help" class="hide">
+                                            Documentation that is hosted in the system and can be edited directly from the UI
```
Comment (harsha89) [inline components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-documents/api-documents.hbs]: Localization need to be add?

File: pom.xml [hunk 9]
```diff
@@ -478,6 +478,11 @@
                 <artifactId>workspace-service</artifactId>
                 <version>0.8.2</version>
             </dependency>
+            <dependency>
```
Comment (harsha89) [inline pom.xml]: any reason for adding this dependency ?

**Comment by pubudu538** [comment]
Based on the commits log it looks like you haven't properly rebased the code. Let's try to avoid merge commits by rebasing the code.

File: components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-details/public/js/details.js [hunk 10]
```diff
@@ -515,6 +515,178 @@ function showTab(tab_name) {
...
+                .addClass("cu-reg-btn btn-delete text-danger doc-listing-delete")
+                .append(icon.clone().removeClass("fw-edit").addClass("fw-delete"));
+            return $('<div></div>').append(edit_button).append(delete_button).html();
+        } else {
+            return data;
+        }
+    }
+
+function initDataTable(raw_data) {
+    console.log(raw_data);
```
Comment (anuruddhal) [inline components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-details/public/js/details.js]: Remove this line.

File: components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-details/public/js/details.js [hunk 11]
```diff
@@ -515,6 +515,178 @@ function showTab(tab_name) {
...
+                 type: $('input[name=optionsRadios]:checked').val(),
+                 summary: $('#summary').val(),
+                 sourceType: $('input[name=optionsRadios1]:checked').val(),
+                 sourceUrl: $('#docUrl').val(),
+                 inlineContent: "string",
+                 otherTypeName: $('#specifyBox').val(),
+                 permission: '[{"groupId" : "1000", "permission" : ["READ","UPDATE"]},{"groupId" : "1001", "permission" : ["READ","UPDATE"]}]',
+                 visibility: "API_LEVEL"
+                 };
+            console.log(api_documents_data);
```
Comment (anuruddhal) [inline components/web/components/org.wso2.carbon.apimgt.publisher.commons.ui/src/main/fragments/api-details/public/js/details.js]: Remove this line.

**Pull request merged**